src/server/scripts/EasternKingdoms/SunkenTemple/sunken_temple.cpp

```diff
@@ -62,6 +63,162 @@ class at_malfurion_stormrage : public AreaTriggerScript
+
+    struct npc_hakkari_bloodkeeperAI : public ScriptedAI
+    {
+        npc_hakkari_bloodkeeperAI(Creature* creature) : ScriptedAI(creature), _instance(*creature->GetInstanceScript()) { }
+
+        void JustEngagedWith(Unit* /*who*/) override
+        {
+            _events.ScheduleEvent(EVENT_CAST_SHADOW_BOLT, 0s, 2s);
+            _events.ScheduleEvent(EVENT_CAST_CORRUPTION, 13s, 17s);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                 return;
+
+            _events.Update(diff);
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_CAST_SHADOW_BOLT:
+                    DoCastVictim(SPELL_SHADOW_BOLT);
+                    _events.ScheduleEvent(EVENT_CAST_SHADOW_BOLT, 3s, 4s);
+                    break;
+                case EVENT_CAST_CORRUPTION:
+                    DoCastVictim(SPELL_CORRUPTION);
+                    _events.ScheduleEvent(EVENT_CAST_CORRUPTION, 11s, 15s);
+                    break;
+                default:
+                    break;
+                }
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+            }
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        EventMap _events;
+        InstanceScript _instance;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetSunkenTempleAI<npc_hakkari_bloodkeeperAI>(creature);
+    }
+};
+
+/*#####
+# npc_nightmare_suppressor
+#####*/
+
+enum NightmareSuppressor
+{
+    EVENT_CAST_SUPPRESSOR = 1,
+    SAY_RANDOM_SPAWN = 0,
+    POINT_ID_AVATAR_OF_HAKKAR = 0
+};
+
+static Position const CenterOfArena = { -467.067f, 273.796f, -90.642f };
+
+class npc_nightmare_suppressor : public CreatureScript
+{
+public:
+    npc_nightmare_suppressor() : CreatureScript("npc_nightmare_suppressor") { }
+
+    struct npc_nightmare_suppressorAI : public ScriptedAI
+    {
+        npc_nightmare_suppressorAI(Creature* creature) : ScriptedAI(creature), instance(creature->GetInstanceScript()) { }
+
+        InstanceScript* instance;
+
+        void DoAction(int32 action) override
+        {
+            switch (action)
+            {
+            case ACTION_CAST_SUPPRESSOR_NIGHTMARE:
+                //me->SetReactState(REACT_PASSIVE);
+                me->SetSpeed(MOVE_RUN, 2);
+                me->GetMotionMaster()->MovePoint(POINT_ID_AVATAR_OF_HAKKAR, CenterOfArena);
+                Talk(SAY_RANDOM_SPAWN);
+                _events.ScheduleEvent(EVENT_CAST_SUPPRESSOR, 12s);
+                break;
+            }
+        }
+
+        void JustEngagedWith(Unit* /*who*/) override
+        {
+            if (Creature* avatar = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AVATAR_OF_HAKKAR)))
```

Comment:
**nopch** error/warning: `sunken_temple.cpp:176:36: fatal error: use of undeclared identifier 'ObjectAccessor'`
```
            if (Creature* avatar = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AVATAR_OF_HAKKAR)))
                                   ^
```
requires `ObjectAccessor.h` in the #include list to have access to `ObjectAccessor` (according to existing script files)